frontend/src/components/common/StarRating/StarRating.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Star from "../Star/Star";
+
+require('../../../assets/css/StarRating.css');
+
+type Props = {
+    text: string;
+    scale: number;
+};
+type State = {
+    rate: number;
+};
+
+class StarRating extends React.Component<Props, State> {
+
+    constructor(props: Props | Readonly<Props>) {
+        super(props);
+        this.state = {
+            rate: 0
+        }
+    }
+
+    checkStar = (starNumber: number) => {
+        this.setState({rate: starNumber});
+    }
+
+    render() {
+        const {text, scale} = this.props;
+        const {rate} = this.state;
+
+        const stars = []
```

Comment:
Nie powinno się w ten sposób pisać logiki. Starts to powinna być zmienna stanowa, a operacja zdefiniowana w pętli for powinna być wyekstraktowana do funkcji i wywoływana w odpowiedniej metodzie lifecycle